app/js/controllers.js

```diff
@@ -0,0 +1,139 @@
+floripaApp.controller("GMapController", ['$scope', 'uiGmapGoogleMapApi', '$location', 'searchService', function ($scope, uiGmapGoogleMapApi, $location, searchService) {
+
+    $scope.goBack = function () {
+        $location.url('/');
+    };
+
+    $scope.search = function () {
+        searchService.streetName = $scope.streetName;
+        searchService.doSearch = true;
+        $location.url('/');
+    };
+
+    uiGmapGoogleMapApi.then(function (maps) {
+
+    });
+
+    $scope.putMarker = function (mapModel, location) {
+        if (!$scope.marker) {
+            $scope.marker = new google.maps.Marker({
+                position: location,
+                map: mapModel
+            });
+        } else {
+            $scope.marker.setPosition(location);
+        }
+    }
+
+    $scope.map = {
+        // should use the one from the device, just a mock
+        center: {
+            latitude: -27.5964796,
```

Comment:
I should get it from the device, but in the case I had to set a value like I did, I would create a properties file and put the values there with a comment explaining where this location is instead of hard-coding like this.